### PR TITLE
PHPStan Level 7 for more DAO PDO

### PR DESCRIPTION
**Pull request by Alkarex**
With new PDO functions to address common type and check problems

### CODE REVIEW

File: lib/Minz/ModelPdo.php [hunk 1]
```diff
@@ -179,4 +179,64 @@ public static function clean() {
...
+		$info = $stm == null ? $this->pdo->errorInfo() : $stm->errorInfo();
+		Minz_Log::error('SQL error ' . $callingFunction . ' ' . json_encode($info));
+		return null;
+	}
+
+	/**
+	 * @param array<string,int|string|null> $values
+	 * @return array<array<string,int|string|null>>|null
+	 */
+	public function fetchAssoc(string $sql, array $values = []): ?array {
```
Comment (Alkarex) [inline lib/Minz/ModelPdo.php:231]: New function(s) that can be used in many situations to handle the several null checks and logs needed when fetching data

File: app/Models/CategoryDAO.php [hunk 2]
```diff
@@ -250,19 +250,12 @@ public function searchById(int $id): ?FreshRSS_Category {
 	/** @return FreshRSS_Category|null|false */
 	public function searchByName(string $name) {
 		$sql = 'SELECT * FROM `_category` WHERE name=:name';
-		$stm = $this->pdo->prepare($sql);
-		if ($stm == false) {
+		$res = $this->fetchAssoc($sql, ['name' => $name]);
```
Comment (Alkarex) [inline app/Models/CategoryDAO.php:253]: Example of use of the new function to shorten code a lot while being stricter

File: app/Controllers/categoryController.php [hunk 3]
```diff
@@ -97,14 +97,16 @@ public function updateAction(): void {
...
 				Minz_Request::bad(_t('feedback.sub.category.not_exist'), $url_redirect);
 			}
 
-			$cat = new FreshRSS_Category($name);
-			$values = array(
+			$values = [
 				'name' => $cat->name(),
-			);
+				'kind' => $cat->kind(),
+				'attributes' => $cat->attributes(),
```
Comment (Alkarex) [inline app/Controllers/categoryController.php:110]: We had a bug there, which was erasing the values of other columns when updating the name of a category.
I am not sure this function is used at all though (the one in `FreshRSS_subscription_Controller::categoryAction()` is used) - to be checked

File: p/api/greader.php [hunk 4]
```diff
@@ -898,13 +898,15 @@ private static function renameTag(string $s, string $dest) {
 			$categoryDAO = FreshRSS_Factory::createCategoryDao();
 			$cat = $categoryDAO->searchByName($s);
 			if ($cat != null) {
-				$categoryDAO->updateCategory($cat->id(), array('name' => $dest));
+				$categoryDAO->updateCategory($cat->id(), [
+					'name' => $dest, 'kind' => $cat->kind(), 'attributes' => $cat->attributes()
```
Comment (Alkarex) [inline p/api/greader.php:902]: We had a bug here: updating the category name would clear its other columns

File: app/Models/CategoryDAO.php [hunk 5]
```diff
@@ -143,7 +143,7 @@ public function addCategoryObject(FreshRSS_Category $category) {
 	}
 
 	/**
-	 * @param array<string,mixed> $valuesTmp
+	 * @param array{'name':string,'kind':int,'attributes':array<string,mixed>} $valuesTmp
```
Comment (Alkarex) [inline app/Models/CategoryDAO.php:146]: Adding this stricter type discovered two bugs (see below)

File: app/Controllers/tagController.php [hunk 6]
```diff
@@ -132,7 +132,7 @@ public function renameAction(): void {
 		$targetTag = $tagDAO->searchByName($targetName);
 		if ($targetTag === null) {
 			// There is no existing tag with the same target name
-			$tagDAO->updateTag($sourceId, ['name' => $targetName]);
+			$tagDAO->updateTagName($sourceId, $targetName);
```
Comment (Alkarex) [inline app/Controllers/tagController.php:135]: To avoid the same bugs than for categories (see below)

**Review by ColonelMoutarde — APPROVED** [review]

**Review by ColonelMoutarde — APPROVED** [review]

**Review by ColonelMoutarde — APPROVED** [review]

**Pull request merged**